build/luajit.m4

```diff
@@ -0,0 +1,110 @@
+dnl -------------------------------------------------------- -*- autoconf -*-
+dnl Licensed to the Apache Software Foundation (ASF) under one or more
+dnl contributor license agreements.  See the NOTICE file distributed with
+dnl this work for additional information regarding copyright ownership.
+dnl The ASF licenses this file to You under the Apache License, Version 2.0
+dnl (the "License"); you may not use this file except in compliance with
+dnl the License.  You may obtain a copy of the License at
+dnl
+dnl     http://www.apache.org/licenses/LICENSE-2.0
+dnl
+dnl Unless required by applicable law or agreed to in writing, software
+dnl distributed under the License is distributed on an "AS IS" BASIS,
+dnl WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+dnl See the License for the specific language governing permissions and
+dnl limitations under the License.
+
+dnl
+dnl luajit.m4: Trafficserver's luajit autoconf macros
+dnl
+
+dnl
+dnl TS_CHECK_LUAJIT: look for luajit libraries and headers
+dnl
+AC_DEFUN([TS_CHECK_LUAJIT], [
+has_luajit=0
+AC_ARG_WITH(luajit, [AC_HELP_STRING([--with-luajit=DIR], [use a specific luajit library])],
+[
+  if test "x$withval" != "xyes" && test "x$withval" != "x"; then
+    luajit_base_dir="$withval"
+    if test "$withval" != "no"; then
+      has_luajit=1
+
+      case "$withval" in
+      *":"*)
+        luajit_include="`echo $withval | sed -e 's/:.*$//'`"
+        luajit_ldflags="`echo $withval | sed -e 's/^.*://'`"
+        AC_MSG_CHECKING(checking for luajit includes in $luajit_include libs in $luajit_ldflags)
+        ;;
+      *)
+	# looking for versioned subdir
+        luajit_include="$withval/include"
+
+        AC_MSG_CHECKING(checking for $luajit_include/luajit-2.0 or $luajit_include/luajit-2.1)
```

Comment:
I think we don't yet support luajit>=2.1.x
Right?

Reply:
I don't know what our official line is, but I know @SolidWallOfCode was trying to build under FC28, which looks to be using luajit 2.1.x.

Reply:
ok... may be then we should be consistent with the pkg check ?

PKG_CHECK_MODULES([LUAJIT], [luajit < 2.1.0, luajit >= 2.0.4]

Right now it is <2.1.0